frontend/post/postService.js

```diff
@@ -3,28 +3,21 @@
 (function() {
     var app = angular.module("app");
 
-    app.service("PostService", function PostService($http, $q, AuthService) {
+    app.service("PostService", function PostService(HttpService, $q, AuthService) {
         var service = this;
         var POSTS_URI = "/api/posts";
         var LIMIT = 10;
         service.posts = [];
         service.user = AuthService.getCurrentUser();
 
         service.get = function getPosts() {
-            var deferred = $q.defer();
-            $http.get("/api/user/timeline").then(function success(response) {
-                service.posts = response.data;
-                deferred.resolve(response);
-            }, function error(response) {
-                deferred.reject(response);
-            });
-            return deferred.promise;
+            return HttpService.get("/api/user/timeline");
         };
 
         service.getNextPosts = function getNextPosts(page) {
             var deferred = $q.defer();
-            $http.get("/api/user/timeline?page=" + page + "&limit=" + LIMIT).then(function success(response) {
-                service.posts = response.data;
+            HttpService.get("/api/user/timeline?page=" + page + "&limit=" + LIMIT).then(function success(response) {
```

Comment:
The HttpService already returns a promise, you can return it.

Reply:
For this situation the service sets its posts property in the success function, that's why it is resolving the promise and returning a new one...